org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/LambdaExpression.java

```diff
@@ -560,9 +560,23 @@ private void analyzeExceptions() {
 		boolean oldAnalyseResources = compilerOptions.analyseResourceLeaks;
 		compilerOptions.analyseResourceLeaks = false;
 		try {
+			int firstLocalLocal = this.scope.outerMostMethodScope().analysisIndex;
+			LocalVariableBinding[] locals = this.scope.locals;
+			if (locals != null) {
+				for (LocalVariableBinding local : locals) {
+					if (local != null) {
+						if (local.isParameter()) {
+							firstLocalLocal = local.id + 1;
+						} else {
+							firstLocalLocal = local.id;
+							break;
```

Comment:
I couldn't find a test touching this block. Do you care to construct one which does?

Reply:
I think you caught a truly unreachable piece of code. When I wrote this I conceptually viewed the lambda's parameters and the any locals of the outermost block body occupying the same scope. But that is now things are laid out. When the lambda has a block body, the block has its own scope - so we will not find any non-parameter true locals in the locals array being traversed. I'll remove this piece of code

Reply:
Now, that you explained this I found a way to bypass your fix (by removing the lambda parameter):
```
import java.io.IOException;
import java.nio.file.Path;
import java.util.function.Consumer;
import java.util.stream.Stream;

public class X11 {

	public X11 parse() throws IOException {
		return null;
	}
	
	Stream<Path> list() throws IOException {
		return null;
	}

	public void foo() throws IOException {

		throwingConsumer(() -> {
			try (final Stream<Path> files = list()) {
				throwingConsumer(() -> {
					final X11 document;// = parse();
					document = parse();
				});
			}
		});
	}
	public static <T, X extends Throwable> ThrowingConsumer<T, X> throwingConsumer(
			final ThrowingConsumer<T, X> consumer) {
		return consumer;
	}
}


interface ThrowingConsumer<T, X extends Throwable> extends Consumer<T> {

	void tryAccept() throws X;

	default void accept(final T t) {
		tryAccept();
	}
}
```
yields
```
		"----------\n" + 
		"1. ERROR in X11.java (at line 19)\n" + 
		"	try (final Stream<Path> files = list()) {\n" + 
		"	                                ^^^^^^\n" + 
		"Unhandled exception type IOException\n" + 
		"----------\n" + 
		"2. ERROR in X11.java (at line 39)\n" + 
		"	tryAccept();\n" + 
		"	^^^^^^^^^^^\n" + 
		"Unhandled exception type X\n" + 
		"----------\n"
```

Reply:
Oops. I'll take a look and follow up. 

But there is already another PR waiting for your review that may completely short circuit the manifestation of a problem due to this.